packages/block-editor/src/store/private-selectors.js

```diff
@@ -288,25 +286,3 @@ export const hasAllowedPatterns = createSelector(
 export function getLastFocus( state ) {
 	return state.lastFocus;
 }
-
-export const getBlockOrderWithAsyncFlag = createSelector(
-	( state, rootClientId ) => {
-		const order = getBlockOrder( state, rootClientId );
-		const selectedBlocks = getSelectedBlockClientIds( state );
-		const visibleBlocks = __unstableGetVisibleBlocks( state );
-		return order.map( ( clientId ) => ( {
-			clientId,
-			// Only provide data asynchronously if the block is
-			// not visible and not selected.
-			isAsync:
-				! visibleBlocks?.has( clientId ) &&
-				! selectedBlocks?.includes( clientId ),
-		} ) );
-	},
-	( state ) => [
-		state.blocks.order,
-		state.selection.selectionStart.clientId,
-		state.selection.selectionEnd.clientId,
-		state.blockVisibility,
```

Comment:
Hm, I guess this is creating a new array when the block visibility changes, but that still doesn't explain the typing regression